legal-api/src/legal_api/models/batch_processing.py

```diff
@@ -103,3 +128,35 @@ def find_by(cls,  # pylint: disable=too-many-arguments
 
         batch_processinges = query.order_by(BatchProcessing.id).all()
         return batch_processinges
+
+    @staticmethod
+    def get_eligible_batch_processings_for_colin(page=1, limit=20):
+        """Return the batch_processings that are ready to sync to colin."""
+        from .batch import Batch  # noqa: F401; pylint: disable=import-outside-toplevel
+        from .colin_event_id import ColinEventId  # noqa: F401; pylint: disable=import-outside-toplevel
+
+        has_colin_event_id = (
+            exists().where(
+                BatchProcessing.id == ColinEventId.batch_processing_id,
+                BatchProcessing.step == ColinEventId.batch_processing_step
+            )
+        )
+        batch_processings = db.session.query(BatchProcessing).\
+            filter(Batch.id == BatchProcessing.batch_id).\
+            filter(Batch.batch_type == Batch.BatchType.INVOLUNTARY_DISSOLUTION).\
+            filter(BatchProcessing.step.in_([
+                BatchProcessing.BatchProcessingStep.WARNING_LEVEL_1,
+                BatchProcessing.BatchProcessingStep.WARNING_LEVEL_2,
+            ])).\
+            filter(BatchProcessing.status == BatchProcessing.BatchProcessingStatus.PROCESSING).\
```

Comment:
Thinking if it's proper to only select batch processing in `PROCESSING` status...?

Reply:
This ticket is on hold for now, but to still answer your question: yes I think it should only select PROCESSING status. My analysis on the statuses is that:
- `ERROR`, `HOLD`, and `QUEUED` statuses we don't want to to sync since there is some issue or wait associated with the batch processing.
- `COMPLETED` status will always have a dissolution filing. This dissolution filing is already synced correctly (creates and EVENT and sets the CORP_STATE).
- `WITHDRAWN` status should always have an annual report filing. This AR filing sync needs to be verified in https://github.com/bcgov/entity/issues/23545

If we were to sync `COMPLETED` and `WITHDRAWN` status batch processings, we would get duplicates in COLIN since we would be syncing both the batch processing and their associated filings.